assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -633,21 +635,31 @@ angular.module('openshiftConsole')
     };
 
     $scope.updateTriggers = function() {
+
+      // Helper for checking webhook objects in the buildConfig spec and creating if missing.
+      function reuseOrCreateWebhook(type) {
+        var webhooks = _.filter($scope.buildConfig.spec.triggers, function(obj) { return obj.type === type})
+        if (_.isEmpty(webhooks)) {
+          webhooks = {
```

Comment:
if webhooks is supposed to be an array, which looking at the later code seems like it should be, then 1) setting webhooks to {type: type} which is an object isn't right, and 2) webhooks["github"] is also not right since that is not an array reference that is a map reference.

i think all you really want here is

```
if (_.isEmpty(webhooks)) {
  var webhook = {
    type: type,
    secret: ApplicationGenerator._generateSecret()
  }
  webhooks.push(webhook);
```